File: BackEnd/WareHouse/Services/WareHouse.Service/Models/GoodsModel.cs
﻿using Microsoft.AspNetCore.Http;
using System;
using System.ComponentModel.DataAnnotations;
using WareHouse.Service.Constants;

namespace WareHouse.Service.Models
{
    /// <summary>
    /// Goods model.
    /// </summary>
    public class GoodsModel
    {
        public string Id { get; set; }

        [Required]
        [MaxLength(20)]
        public string Code { get; set; }

        [Required]
        [MaxLength(200)]
        public string Name { get; set; }

        [MaxLength(200)]
        public string Brand { get; set; }

        [MaxLength(200)]
        public string Color { get; set; }

        [MaxLength(200)]
        public string Size { get; set; }

        public IFormFile File { get; set; }

        public string FileId { get; set; }

        public string FileContent { get; set; }

        public string Description { get; set; }

        [Required]
        public string UnitId { get; set; }

        public string UnitName { get; set; }

        [Required]
        public string GoodsCategoryId { get; set; }

        public string GoodsCategoryName { get; set; }

        [Required]
        public string IsActive { get; set; } = "1";
        
        public byte[] RowVersion { get; set; }

        public string CurrentUserId { get; set; }

        public string IsEdit { get; set; } = Status.Insert;
    }
}

File: FrontEnd/warehouse/src/app/modules/warehouse/goods/models/goods.model.ts

export class GoodsModel {
  id: string;
  code: string;
  name: string;
  brand: string;
  color: string;
  size: string;
  fileId: string;
  fileContent: string;
  fileName: string;
  description: string;
  unitId: string;
  unitName: string;
  goodsCategoryId: string;
  goodsCategoryName: string;
  isEdit: boolean;
  isActive: boolean;
  rowVersion: any;

  constructor(isEdit: boolean) {
    this.id = '';
    this.code = '';
    this.name = '';
    this.brand = '';
    this.color = '';
    this.size = '';
    this.fileId = '';
    this.fileContent = '';
    this.fileName = '';
    this.description = '';
    this.unitId = '';
    this.unitName = '';
    this.goodsCategoryId = '';
    this.goodsCategoryName = '';
    this.isEdit = isEdit;
    this.isActive = true;
  }
}
